src/app/modules/main/wallet_section/activity/controller.nim

```diff
@@ -291,4 +301,22 @@ QtObject:
 
   QtProperty[int] errorCode:
     read = getErrorCode
-    notify = errorCodeChanged
\ No newline at end of file
+    notify = errorCodeChanged
+
+  proc updateRecipientsModel*(self: Controller) {.slot.} =
+    let response = backend_activity.getAllRecipients(0, FETCH_RECIPIENTS_BATCH_COUNT_DEFAULT)
+    if response.error != nil:
+      error "error fetching recipients: ", response.error
+      return
+
+    let result = json.to(response.result, backend_activity.GetAllRecipientsResponse)
+    self.recipientsModel.addAddresses(result.addresses, 0, result.hasMore)
+
+  proc loadMoreRecipients(self: Controller) {.slot.} =
```

Comment:
Hey @stefandunca why is this needed?
will the user only get list of recipient based on transactions loaded?
Also will they be exlusive and non repeating?

Reply:
Because we can't control how many transactions might be in the DB, there is no limit to how many recipients we can have, so this way, we set a limit to the retrieve (2000 for now) and have the option to retrieve the rest later on. Having something more than lazy loading of batches was too much. However, if having more than a few hundred is expected and not an extreme corner case, we can extend this with search and lazy loading like the activity history.

For now, don't use it. Load the first batch, and let's see if that is enough (based on testing with famous accounts). The next improvement will be retrieving asynchronous, leaving it for the next cycle.